lib/skate_web/channels/user_socket.ex

```diff
@@ -15,8 +15,11 @@ defmodule SkateWeb.UserSocket do
   #
   # See `Phoenix.Token` documentation for examples in
   # performing token verification on connect.
-  def connect(_params, socket, _connect_info) do
-    {:ok, socket}
+  def connect(%{"token" => token}, socket, _connect_info) do
```

Comment:
Great. The other thing to consider is whether and how often to verify the token is still valid. For instance, in our signs_ui channel that turns signs on/off and puts custom text on them we [verify the token](https://github.com/mbta/signs_ui/blob/de732233f1f6cc94ade4073973b176b9f762d64a/lib/signs_ui_web/channels/signs_channel.ex#L12) on the channel `handle_in` callback. And if it's no longer valid, we respond with `"auth_expired"` and then close the websocket. The frontend has a bit of logic to detect `"auth_expired"` and refresh the page (which, because of `:ensure_auth` and Guardian, will redirect to the Cognito sign in page).

Reply:
Nice, copied this shamelessly.